MobiFlight/Monitors/UsbDeviceMonitor.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace MobiFlight.Monitors
+{
+    public class UsbDeviceMonitor : DeviceMonitor
+    {
+        /// <summary>
+        /// Returns a list of connected USB drives that are supported with MobiFlight and are in flash mode already,
+        /// as opposed to being connected as COM port.
+        /// </summary>
+        /// <returns>The list of connected USB drives supported by MobiFlight.</returns>
+        override protected void ScanPorts()
+        {
+            var result = new List<PortDetails>();
+            
+            foreach (var drive in DriveInfo.GetDrives())
+            {
+                // Issue 1074: Failing to check for IsReady caused an IOException on certain machines
```

Comment:
The refactor of this into a new class is cool but it lost an important bug fix from the old code:

```C#
// Issue 1089: Network drives take *forever* to return the drive info slowing down startup. Only check removable drives.
                if (drive.DriveType != DriveType.Removable)
                {
                    Log.Instance.log($"Drive {drive.Name} ({drive.DriveType}) isn't a removable drive, skipping.", LogSeverity.Debug);
                    continue;
                }
```